sdk/api_fingerprint.go

```diff
@@ -11,328 +11,118 @@ package sdk
-		// API Key Authentication
-		if auth, ok := ctx.Value(ContextAPIKey).(APIKey); ok {
-			var key string
-			if auth.Prefix != "" {
-				key = auth.Prefix + " " + auth.Key
-			} else {
-				key = auth.Key
+			for i, arg := range args {
+				path = strings.Replace(path, "{"+pathParams[i]+"}", arg, -1)
 			}
-			localVarHeaderParams["Auth-API-Key"] = key
-		}
-	}
-	r, err := a.client.prepareRequest(ctx, localVarPath, localVarHttpMethod, localVarPostBody, localVarHeaderParams, localVarQueryParams, localVarFormParams, localVarFileName, localVarFileBytes)
-	if err != nil {
-		return localVarReturnValue, nil, err
-	}
 
-	localVarHttpResponse, err := a.client.callAPI(r)
-	if err != nil || localVarHttpResponse == nil {
-		return localVarReturnValue, localVarHttpResponse, err
+			return path
+		},
+		StatusCodeResultsFactoryMap: map[int]func() interface{}{
+			403: func() interface{} { return &ErrorCommon403Response{} },
+			404: func() interface{} { return &ErrorVisitsDelete404Response{} },
+		},
 	}
+}
 
-	localVarBody, err := io.ReadAll(localVarHttpResponse.Body)
-	localVarHttpResponse.Body.Close()
-	if err != nil {
-		return localVarReturnValue, localVarHttpResponse, err
-	}
-
-	if localVarHttpResponse.StatusCode < 300 {
-		// If we succeed, return the data, otherwise pass on to decode error.
-		err = a.client.decode(&localVarReturnValue, localVarBody, localVarHttpResponse.Header.Get("Content-Type"))
-
-		ParseResponse(localVarHttpResponse, &err)
-
-		return localVarReturnValue, localVarHttpResponse, err
-	}
-
-	if localVarHttpResponse.StatusCode >= 300 {
-		newErr := GenericSwaggerError{
-			body:  localVarBody,
-			error: localVarHttpResponse.Status,
-		}
-		if localVarHttpResponse.StatusCode == 200 {
-			var v Response
-			err = a.client.decode(&v, localVarBody, localVarHttpResponse.Header.Get("Content-Type"))
-			if err != nil {
-				newErr.error = err.Error()
-
-				ParseResponse(localVarHttpResponse, &newErr)
-
-				return localVarReturnValue, localVarHttpResponse, newErr
-			}
-			newErr.model = v
-
-			ParseResponse(localVarHttpResponse, &newErr)
-
-			return localVarReturnValue, localVarHttpResponse, newErr
-		}
-		if localVarHttpResponse.StatusCode == 403 {
-			var v ErrorVisits403
-			err = a.client.decode(&v, localVarBody, localVarHttpResponse.Header.Get("Content-Type"))
-			if err != nil {
-				newErr.error = err.Error()
+func createGetEventDefinition() requestDefinition {
+	return requestDefinition{
+		Path: func(args ...string) string {
+			pathParams := []string{"request_id"}
 
-				ParseResponse(localVarHttpResponse, &newErr)
+			path := "/events/{request_id}"
 
-				return localVarReturnValue, localVarHttpResponse, newErr
+			for i, arg := range args {
+				path = strings.Replace(path, "{"+pathParams[i]+"}", arg, -1)
 			}
-			newErr.model = v
 
-			ParseResponse(localVarHttpResponse, &newErr)
+			return path
+		},
+		StatusCodeResultsFactoryMap: map[int]func() interface{}{
+			200: func() interface{} { return &EventResponse{} },
```

Comment:
Could you show how the case `200` is used? I can only find the usage inside `handleErrorResponse` function. If it is really not used, let's remove it to avoid confusion.